src/Sylius/Behat/Context/Setup/PromotionContext.php

```diff
@@ -564,6 +565,22 @@ public function thePromotionGivesOffIfOrderContainsProducts(PromotionInterface $
     }
 
     /**
+     * @Given /^([^"]+) gives ("(?:€|£|\$)[^"]+") off on a ("[^"]*" product)$/
+     */
+    public function itGivesFixedDiscountOffOnAProduct(PromotionInterface $promotion, $discount, ProductInterface $product)
+    {
+        $this->createUnitFixedPromotion($promotion, $discount, $this->getProductsFilterConfiguration([$product->getCode()]));
+    }
+
+    /**
+     * @Given /^([^"]+) gives ("[^"]+%") off on a ("[^"]*" product)$/
+     */
+    public function itGivesPercentageDiscountOffOnAProduct(PromotionInterface $promotion, $discount, ProductInterface $product)
+    {
+        $this->createUnitPercentagePromotion($promotion, $discount, $this->getProductsFilterConfiguration([$product->getCode()]));
```

Comment:
This promotion will be applied to every order, but not every order will be affected by it's action. We should only apply the promotion to orders that contain this product, so we should add a promotion rule `ContainsProduct` and probably modify the step to "it gives "20%" off on a "PHP T-Shirt" product **if order contains it**"